File: KampIntro_Odevler/Gun_05_Odev_05/Gun_05_Odev_05/Abstract/BaseCampaignManager.cs
using Gun_05_Odev_05.Entities;
using System;
using System.Collections.Generic;
using System.Text;

namespace Gun_05_Odev_05.Abstract
{
    public class BaseCampaignManager : ICampaignService
    {
        public virtual void Add(Campaign campaign)
        {
            Console.WriteLine("Kampanya veri tabanına eklendi : " + campaign.CampaignName + " " + campaign.CampaignDiscountRate);
        }

        public virtual void Delete(Campaign campaign)
        {
            Console.WriteLine("Kampanya silindi : " + campaign.CampaignName + " " + campaign.CampaignDiscountRate);
        }

        public virtual void Update(Campaign campaign)
        {
            Console.WriteLine("Kampanya güncellendi : " + campaign.CampaignName + " " + campaign.CampaignDiscountRate);
        }
    }
}

File: Java_React_Ödevler/Java_React_Ödev_Gün_04_Ödev_03_OyunSatis/Gun_04_Odev_03/Abstract/BaseCampaignManager.java

public abstract class BaseCampaignManager implements CampaignService {
	
	public void add(Campaign campaign)
    {
        System.out.println("Kampanya veri tabanına eklendi : " + campaign.getName() + " " + campaign.getDiscountRate());
    }

    public void delete(Campaign campaign)
    {
    	System.out.println("Kampanya veri tabanından silindi : " + campaign.getName() + " " + campaign.getDiscountRate());
    }

    public void update(Campaign campaign)
    {
    	System.out.println("Kampanya bilgileri güncellendi : " + campaign.getName() + " " + campaign.getDiscountRate());
    }
}